src/target/riscv/riscv-013_reg.c

```diff
@@ -92,26 +93,165 @@ static const struct reg_arch_type *riscv013_gdb_regno_reg_type(uint32_t regno)
 	return &riscv011_reg_type;
 }
 
-static int riscv013_init_reg(struct target *target, uint32_t regno)
+static int init_cache_entry(struct target *target, uint32_t regno)
 {
-	return riscv_reg_impl_init_one(target, regno, riscv013_gdb_regno_reg_type(regno));
+	return riscv_reg_impl_init_cache_entry(target, regno, riscv013_gdb_regno_reg_type(regno));
 }
 
-int riscv013_reg_init_all(struct target *target)
+static int examine_xlen(struct target *target)
 {
-	if (riscv_reg_impl_init_cache(target) != ERROR_OK)
+	RISCV_INFO(r);
+	unsigned int cmderr;
+	const uint32_t command = riscv013_access_register_command(target,
+			GDB_REGNO_S0, /* size */ 64, AC_ACCESS_REGISTER_TRANSFER);
+	int res = riscv013_execute_abstract_command(target, command, &cmderr);
+	if (res == ERROR_OK) {
+		r->xlen = 64;
+		return ERROR_OK;
+	}
+	if (res == ERROR_TIMEOUT_REACHED)
 		return ERROR_FAIL;
+	r->xlen = 32;
+	return ERROR_OK;
```

Comment:
```suggestion
	RISCV_INFO(r);
	unsigned int cmderr;
	
	const uint32_t command = riscv013_access_register_command(target,
			GDB_REGNO_S0, /* size */ 64, AC_ACCESS_REGISTER_TRANSFER);
	int res = riscv013_execute_abstract_command(target, command, &cmderr);
	if (res == ERROR_OK) {
		r->xlen = 64;
		return ERROR_OK;
	}
	if (res == ERROR_TIMEOUT_REACHED)
		return ERROR_FAIL;
	r->xlen = 32;
	
	return ERROR_OK;
```
Detail: I believe it would help readability.

Reply:
Addressed.